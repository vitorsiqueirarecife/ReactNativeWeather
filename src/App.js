import React, { useEffect } from 'react';
import { TouchableOpacity, SafeAreaView, View, Text } from 'react-native';
import { style } from './styles';
import useLocation from './Hooks/useLocation';
import useWeather from './Hooks/useWeather'

const App = () => {

  const { loadPosition, coordenates} = useLocation();	
  const { getWeather, informations } = useWeather();
  
  useEffect(()=>{
    const run = async () => {
      await loadPosition()
    }
    run()
  },[])

  useEffect(()=>{
    const run = async () => {
       await getWeather(coordenates?.latitude, coordenates?.longitude)     
    }
    run()
  },[coordenates])

  return (
      <SafeAreaView style={style.wrapper}>
          <View style={style.card}>
            <Text style={style.title}>Location</Text>
          </View>
          <View style={style.cardContent}>
            <Text style={style.content}>{informations?.name} - {informations?.sys?.country}</Text>
          </View>
          <View style={style.card}>
            <Text style={style.title}>Weather</Text>
          </View>
          <View style={style.cardContent}>
            <Text style={style.content}>{informations?.main?.temp} graus celsius ({informations?.weather?.[0]?.description})</Text>
            <Text style={style.content}>Humidity: {informations?.main.humidity}</Text>
            <Text style={style.content}>Pressure: {informations?.main.pressure}</Text>  
          </View>            
          <TouchableOpacity style={style.button} >
            <Text style={style.title} onPress={()=> {
              getWeather(coordenates?.latitude, coordenates?.longitude)   
            }} >Atualizar</Text>
          </TouchableOpacity>                 
      </SafeAreaView>
  );
};

export default App;
