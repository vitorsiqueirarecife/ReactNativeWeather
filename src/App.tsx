import React from 'react';
import Weather from './modules/weather/screens'
import { NavigationContainer } from '@react-navigation/native';
import { createNativeStackNavigator } from '@react-navigation/native-stack';
import Home from './modules/home/screens';

const Stack = createNativeStackNavigator();

const App = () => {

  return (
    <NavigationContainer>
      <Stack.Navigator
         screenOptions={{
          headerShown: false          
        }}
      >
        <Stack.Screen name="Home" component={Home} />
        <Stack.Screen name="Weather" component={Weather} />
      </Stack.Navigator>
    </NavigationContainer>
  );
};

export default App;
