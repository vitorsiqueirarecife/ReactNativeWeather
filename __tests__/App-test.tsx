import 'react-native';
import  React from 'react';
import App from '../src/App';
import renderer from 'react-test-renderer';
// import {fireEvent, render} from '@testing-library/react-native';

jest.mock('react-native-permissions', () => ({
  check: (_: boolean) => Promise.resolve(true),
}));

it('renders correctly', () => {
  const tree = renderer.create(<App />).toJSON();
  expect(tree).toMatchSnapshot();
});

// it('should render default components', () => {
//   const {getByText} = render(<App />);

//   getByText('Seja bem-vindo!');
//   getByText('Por favor, aguarde');
// });

// it('should add one more on array when press count button', () => {
//   const {getByTestId, getByText} = render(<App />);

//   fireEvent.press(getByTestId('countButton'));
//   fireEvent.press(getByTestId('countButton'));

//   getByText('(0)');
//   getByText('(1)');
//   getByText('(2)');
// });
