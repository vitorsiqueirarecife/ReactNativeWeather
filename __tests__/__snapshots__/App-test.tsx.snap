// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<RNCSafeAreaProvider
  onInsetsChange={[Function]}
  style={
    Array [
      Object {
        "flex": 1,
      },
      undefined,
    ]
  }
>
  <RNSScreenStack
    style={
      Object {
        "flex": 1,
      }
    }
  >
    <RNSScreen
      collapsable={false}
      forwardedRef={[Function]}
      onAppear={[Function]}
      onDisappear={[Function]}
      onDismissed={[Function]}
      onTransitionProgress={[Function]}
      onWillDisappear={[Function]}
      replaceAnimation="push"
      stackPresentation="push"
      style={
        Object {
          "bottom": 0,
          "left": 0,
          "position": "absolute",
          "right": 0,
          "top": 0,
        }
      }
    >
      <RNSScreenStackHeaderConfig
        backButtonInCustomView={false}
        backgroundColor="rgb(255, 255, 255)"
        color="rgb(0, 122, 255)"
        direction="ltr"
        disableBackButtonMenu={false}
        hidden={true}
        hideBackButton={false}
        largeTitleHideShadow={false}
        title="Home"
        titleColor="rgb(28, 28, 30)"
        topInsetEnabled={false}
        translucent={false}
      />
      <View
        style={
          Array [
            Object {
              "flex": 1,
            },
            Object {
              "backgroundColor": "rgb(242, 242, 242)",
            },
            undefined,
          ]
        }
      >
        <RCTSafeAreaView
          className="sc-eCImPb hblKfd"
          emulateUnlessSupported={true}
        >
          <View
            className="sc-gKclnd jwUjax"
          >
            <Text
              className="sc-iCfMLu glJgIk"
            >
              Seja bem-vindo!
            </Text>
          </View>
          <View
            className="sc-gKclnd jwUjax"
          >
            <Text
              className="sc-furwcr oKJML"
            >
              Por favor, aguarde
            </Text>
          </View>
          <View
            className="sc-gKclnd jwUjax"
          >
            <ActivityIndicator
              color="#8c87d4"
              size="large"
            />
          </View>
        </RCTSafeAreaView>
      </View>
    </RNSScreen>
  </RNSScreenStack>
</RNCSafeAreaProvider>
`;
